src/sqlfluff/dialects/dialect_snowflake.py

```diff
@@ -144,6 +144,14 @@
         name="warehouse_size",
         type="warehouse_size",
     ),
+    # We use a RegexParser instead of keywords as the arguments are optionally quoted.
+    CompressionType=RegexParser(
+        r"'?AUTO'?|'?GZIP'?|'?BZ2'?|'?BROTLI'?|'?ZSTD'?|'?DEFLATE'?|'?RAW_DEFLATE'?|"
+        r"'?LZO'?|'?NONE'?|'?SNAPPY'?",
```

Comment:
Slightly more efficient and easier to read regex:

```suggestion
        r"'?(AUTO|GZIP|BZ2|BROTLI|ZSTD|DEFLATE|RAW_DEFLATE|LZO|NONE|SNAPPY)'?",
```

The down side to this is it will accept unbalanced quotes (e.g. `'ANSI`).

To avoid this, you could have this as a `OneOf`:

```py
    CompressionType=OneOf(
         RegexParser( r"'(AUTO|GZIP|...)'",....
         RegexParser( r"(AUTO|GZIP|...)",....
    ),
```

Reply:
I'd probably suggest exploding this out to all of the options in StringParser. I dislike RegexParser, as it's really not performant - and I'd make the case for not using it anywhere

Reply:
In thought you might say this, cause I know you've raised this before. 😁

But is it really that badly performant? Especially for simple regexes like this? Realistically we're not going to get away from the RegexParser for lots of things and do think the StringParser leads to a lot more verbose code for things like this.

Btw, think I've spotted a small optimisation there so will open a separate PR and send to you to review.

Reply:
Actually ignore that last comment. I don't have an optimisation after all.

Reply:
Gone with your suggestion 👍